Dart/src/com/jetbrains/lang/dart/analyzer/DartServerData.java

```diff
@@ -181,6 +184,34 @@ void computedImplemented(@NotNull final String filePath,
   }
 
   @NotNull
+  List<DartError> getErrors(@NotNull final Module module) {
+    final List<DartError> errors = new ArrayList<>();
+
+    for (String path : myErrorData.keySet()) {
+      final VirtualFile file = LocalFileSystem.getInstance().findFileByPath(path);
+      if (file != null && ModuleUtilCore.moduleContainsFile(module, file, false)) {
+        errors.addAll(myErrorData.get(path));
```

Comment:
`myErrorData` is accessed in multiple threads. I'm going to surround the whole loop with `synchronized(myErrorData)`.